cmd/dcrtime/config.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"path/filepath"
+
+	"github.com/decred/dcrd/dcrutil/v2"
+
+	flags "github.com/jessevdk/go-flags"
+)
+
+const (
+	defaultConfigFilename = "dcrtime.conf"
+)
+
+var (
+	defaultHomeDir    = dcrutil.AppDataDir("dcrtime", false)
+	defaultConfigFile = filepath.Join(defaultHomeDir, defaultConfigFilename)
+)
+
+// config defines the configuration options for dcrtime.
+//
+// See loadConfig for details on the configuration load process.
+type config struct {
+	APIToken string `long:"apitoken" description:"Token for accessing privileged API resources"`
+}
+
+// loadConfig initializes and parses the config using a config file
+func loadConfig() (*config, error) {
+	// Default config.
+	cfg := config{
+		APIToken: "",
+	}
+
+	err := flags.IniParse(defaultConfigFile, &cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	err = initHomeDirectory(defaultHomeDir)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cfg, nil
+}
+
+// initHomeDirectory creates the home directory if it doesn't already exist.
```

Comment:
You need to expand and clean the path before creating anything. Look at politeia, it has a function called expandAndCleanPath (or something). Copy-paste it into this code and call it on all paths that are created on disk.

Reply:
done. also fixed an issue where the ini was loaded before the home dir was created.